config/clouddriver.yml

```diff
@@ -40,3 +40,20 @@ google:
     - name: ${providers.google.primaryCredentials.name}
       project: ${providers.google.primaryCredentials.project}
       jsonPath: ${providers.google.primaryCredentials.jsonPath}
+
+cf:
+  enabled: ${providers.cf.enabled:false}
+
+  # TODO: Migrate this out of clouddriver and into either orca or igor
+  jenkins:
+    username: ${services.jenkins.defaultMaster.username}
+    password: ${services.jenkins.defaultMaster.password}
+
+  accounts:
+    - name: ${providers.cf.primaryCredentials.name}
+      api: ${providers.cf.primaryCredentials.api}
+      console: ${providers.cf.primaryCredentials.console}
+      org: ${providers.cf.defaultOrg}
+      space: ${providers.cf.defaultSpace}
+      username: ${cf.account.name}
```

Comment:
I'm not a fan of environ variables at all but recognize they are common and
accepted practice.
I'm just noting the convention that is in use to keep things more
readable/predictable in the "global" namespace.
Especially since the possible sources of these values are sharded to the
wind.
Once you drop into a provider, you can (and should be able to) follow the
providers conventions or idiosyncrasies.


On Fri, Nov 20, 2015 at 4:11 PM, Greg Turnquist <notifications@github.com>
wrote:

> In config/clouddriver.yml
> <https://github.com/spinnaker/spinnaker/pull/557#discussion_r45518893>:
>
> > +
> > +cf:
> > +  enabled: ${providers.cf.enabled:false}
> > +
> > +  # TODO: Migrate this out of clouddriver and into either orca or igor
> > +  jenkins:
> > +    username: ${services.jenkins.defaultMaster.username}
> > +    password: ${services.jenkins.defaultMaster.password}
> > +
> > +  accounts:
> > +    - name: ${providers.cf.primaryCredentials.name}
> > +      api: ${providers.cf.primaryCredentials.api}
> > +      console: ${providers.cf.primaryCredentials.console}
> > +      org: ${providers.cf.defaultOrg}
> > +      space: ${providers.cf.defaultSpace}
> > +      username: ${cf.account.name}
>
> FYI I'm not a fan of declaring UPPER_CASE env variables, when every single
> property you're seeing support overrides via the same method.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/spinnaker/spinnaker/pull/557/files#r45518893>.
>